### PR TITLE
Added installation management APIs to notification hub nodejs SDK

### PR DESCRIPTION
**Pull request by ankurkul**
Notification hub team: Added installation management APIs to SDK. Added CIT tests as well.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@ankurkul__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

File: lib/common/lib/util/constants.js [hunk 1]
```diff
@@ -2684,7 +2684,8 @@ var Constants = {
       DELETE: 'DELETE',
       POST: 'POST',
       MERGE: 'MERGE',
-      HEAD: 'HEAD'
+      HEAD: 'HEAD',
+	  PATCH: 'PATCH'
```
Comment (yugangw-msft) [inline lib/common/lib/util/constants.js]: indent

File: test/services/serviceBus/installation-management-tests.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        suiteUtil.setupSuite(done);
+    });
+    
+    after(function (done) {
+        sandbox.restore();
+        suiteUtil.teardownSuite(done);
+    });
+    
+    beforeEach(function (done) {
+        suiteUtil.setupTest(function () {
```
Comment (yugangw-msft) [inline test/services/serviceBus/installation-management-tests.js]: indent at 2

File: lib/services/serviceBus/lib/notificationhubservice.js [hunk 3]
```diff
@@ -434,6 +434,142 @@ NotificationHubService.prototype._sendNotification = function (payload, optionsO
...
+/**
+* Creates or updates a installation.
+*
+* @param {string}             installation              The installation to create/update.
+* @param {object}             [options]                 The request options or callback function. Additional properties will be passed as headers.
+* @param {Function(error, response)} callback           `error` will contain information
+*                                                       if an error occurs; otherwise, `response`
+*                                                       will contain information related to this operation.
+*/
+NotificationHubService.prototype.createOrUpdateInstallation = function (installation, optionsOrCallback, callback) {
```
Comment (mamaso) [inline lib/services/serviceBus/lib/notificationhubservice.js]: The installation param looks to be an object, not a string

**Pull request merged**